dataprofiler/data_readers/json_data.py

```diff
@@ -392,14 +394,16 @@ def is_match(
                 return True
             except (json.JSONDecodeError, UnicodeDecodeError):
                 data_file.seek(0)
-
+            json_identifier_re = re.compile(r"(:|\[)")
```

Comment:
allows both `:` and `[` as a JSON identifier differentiating from just a string.